flowmachine/flowmachine/core/connection.py

```diff
@@ -356,8 +362,8 @@ def available_dates(
         """
 
         if isinstance(table, str) and table.lower() == "all":
-            table = self.subscriber_tables
-        tables = table if isinstance(table, list) else [table]
+            table = tuple(self.subscriber_tables)
+        tables = table if isinstance(table, tuple) else [table]
```

Comment:
Since we're doing type checking anyway, do we want to add a check that `table` can only be a string or tuple of strings? E.g.:
```python
if not isinstance(table, str) or not isinstance(table, tuple) or not all([isinstance(x, str) for x in table]):
    raise TypeError(f"Argument 'table' must be a string or tuple of strings. Got: {table}")
```